File: lab1/I.py
x = int(input())
s = input()
a = []

for i in s:
    a.append(i)

k = 0
s = 0


while True:

    if k>len(a) or s>len(a):
        break

    if a[0] == 'S' and s == 0:
        k+=1
        a.pop(0)
        a.append('S')
        
    elif a[0] == 'K' and k == 0:
        s+=1
        a.append('K')
        a.pop(0)
    elif a[0] == 'K' and k != 0:
        k-=1
        a.pop(0)
    elif a[0] == 'S' and s != 0:
        s-=1
        a.pop(0)
print(a)

if a[-1] == 'S':
    print("SAKAYANAGI")
else:
    print("KATSURAGI")





       
        


    

File: lab1/I.cpp
#include <bits/stdc++.h>

using namespace std;




int main () {
    int n;
    int k = 0;
    int s = 0;
    cin >> n;
    string st;
    cin >> st;
    queue <char> ks;


    for (int i = 0; i < st.size(); i++){
            ks.push(st[i]);
        }

    while (k > ks.size() || s > ks.size() ){
        
        if (ks.front() == 'S' and s == 0){
            k+=1;
            ks.push('S');
            ks.pop();
               
        }
        else if(ks.front() == 'K' && k == 0){
            s+=1;
            ks.push('K');
            ks.pop();
            
            
            
            
        }
        else if(ks.front() == 'K' && k != 0){
            k-=1;
            ks.pop();
            
            
        }
        else if(ks.front() == 'S' && s != 0){
            s-=1;
            ks.pop();
            
        }

    }

   if(ks.front()=='S'){
    cout<<"SAKAYANAGI";
   }
   else if(ks.front()=='K')
   cout<<"KATSURAGI";
    
}